src/app/services/auth-state-change.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/auth';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthStateChangeService {
+  userData: any; // logged in user data
```

Comment:
Don't use any as data - see if we can create an object so that we can map to the userData

Reply:
can't use the interface I created as it needs to be of type firebase.User, which for some reason cannot be defined here..